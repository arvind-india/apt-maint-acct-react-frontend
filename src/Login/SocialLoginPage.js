import React from 'react'
import { connect } from 'react-redux'
import { Button  } from 'reactstrap'
import {
          FaFacebook,
          FaGoogle,
          FaGithub
} from 'react-icons/lib/fa' // material design icons

import { userActions, alertActions } from '../_actions'

let hello = require('hellojs/dist/hello.all.js')

hello.init({
    facebook: process.env.REACT_APP_FACEBOOK_CLIENT_ID,
    google: process.env.REACT_APP_GOOGLE_CLIENT_ID,
    github: process.env.REACT_APP_GITHUB_CLIENT_ID
  }, {
    scope: 'email'
  }
)

export class SocialLogin extends React.Component {

  constructor(props) {
    super(props)
    // reset login status
    // this.props.dispatch(userActions.logout())
    // this.props.dispatch(alertActions.clear())  // clear alert messages from other pages

    this.props.logout()
    this.props.clearAlert()
  }

  render() {
    return (
      <i className="float-right">{this.fbButton()}.{this.googleButton()}</i>
    )
  }

  googleButton() {
    return <Button
      color="link"
      onClick={()=>this.socialLogin('google')}
      title="Google login"
    ><FaGoogle/></Button>
  }

  fbButton() {
    return <Button
      color="link"
      onClick={()=>this.socialLogin('facebook')}
      title="Facebook login"
    ><FaFacebook/></Button>
  }

  githubButton() {
    return <Button
      color="link"
      onClick={()=>this.socialLogin2('github')}
      title="github"
    ><FaGithub/></Button>
  }

  socialLogin(network){
    // const { dispatch } = this.props

    let hello = require('hellojs/dist/hello.all.js')

    hello.init({
        facebook: process.env.REACT_APP_FACEBOOK_CLIENT_ID,
        google: process.env.REACT_APP_GOOGLE_CLIENT_ID,
        github: process.env.REACT_APP_GITHUB_CLIENT_ID
      }, {
        scope: 'email'
      }
    )

    hello(network).login().then(socialResponse, error)

    function socialResponse() {
      let authResponse = hello(network).getAuthResponse();
      let token = authResponse.access_token;
      // dispatch( userActions.socialLogin(network, token) )
      this.props.socialLogin(network, token)
    }
    function error(err) {
      console.error('Social Login Error: ', err)
    }
  }

}


function mapStateToProps(state) {
  const { alert } = state
  const { loggingIn } = state.authentication
  return {
    alert,
    loggingIn
  }
}

function mapDispatchToProps(dispatch) {
  return {
    socialLogin: (email, password) => {
      dispatch(userActions.socialLogin(email, password))
    },
    logout: () => {
      dispatch(userActions.logout())
    },
    clearAlert: () => {
      dispatch(alertActions.clear())
    }
  }
}

// const connectedSocialLoginPage = connect(mapStateToProps)(SocialLoginPage)
// export { connectedSocialLoginPage as SocialLoginPage }

export default connect(mapStateToProps, mapDispatchToProps)(SocialLogin)
