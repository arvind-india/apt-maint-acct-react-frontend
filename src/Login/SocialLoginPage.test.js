// SocialLoginPage.test.js

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';

import { SocialLogin } from './SocialLoginPage'

const mockProps = {
  // alert: {message: "testingAlert"},
  //authentication: {loggingIn: true, user: {}},
  socialLogin: jest.fn(),  // mock function
  logout: jest.fn(), // mock function
  clearAlert: jest.fn() // mock function
}
/*
const mockStates = {
  email: 'unitTesting@jest.js',
  password: 'secret'
}
const mockEmail = {
  name: 'email',
  value: "dummy@email.id"
}
const mockPassword = {
  name: 'password',
  value: "secret"
}
const mockResult = [
  mockEmail.value,
  mockPassword.value
] */
const component = shallow(<SocialLogin {...mockProps} />)

describe('SocialLoginPage params test', () => {
  it('should render SocialLoginPage without crashing', () => {
    component
  })
  it('should render initial layout', () => {
    expect(component.getElements()).toMatchSnapshot()
  })
/*  it('should have email state', () => {
    expect(component.state('email')).toBeDefined()
  })
  it('should have password state', () => {
    expect(component.state('password')).toBeDefined()
  })
  it('should set email state', () => {
    component.setState({email: mockStates.email})
    expect(component.state('email')).toEqual(mockStates.email)
  })
  it('should set password state', () => {
    component.setState({password: mockStates.password})
    expect(component.state('password')).toEqual(mockStates.password)
  }) */
})

/*
describe('SocialLoginPage events test', () => {
  it('should call mock login function', () => {
    component.find('#loginForm').simulate('submit', {preventDefault() {}})
    expect(mockProps.login.mock.calls.length).toEqual(1)
  })
  it('should be called with the email and password in the state as arguments', () => {
    component.find('#email').simulate('change', {target: mockEmail})
    component.find('#password').simulate('change', {target: mockPassword})
    component.find('#loginForm').simulate('submit', {preventDefault() {}})
    expect(mockProps.login.mock.calls[1]).toEqual(mockResult)
  })
})
*/
