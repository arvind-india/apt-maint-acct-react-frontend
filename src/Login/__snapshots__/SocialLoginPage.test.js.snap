// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLoginPage params test should render initial layout 1`] = `
Array [
  <i
    className="float-right"
>
    <Button
        color="link"
        onClick={[Function]}
        tag="button"
        title="Facebook login"
    >
        <FaFacebook />
    </Button>
    .
    <Button
        color="link"
        onClick={[Function]}
        tag="button"
        title="Google login"
    >
        <FaGoogle />
    </Button>
</i>,
]
`;
